cypress-io/cypress
# fix: remove dependence on @cypress/<dep> types

- Closes #24330

### User facing changelog
Fix types associated with `cy.mount` that we incorrectly `any` typed

### Additional details
When importing from `cypress/<framework>`, the types for the mount options were incorrectly typed as `any` if the option type was dependent on another cypress package such as `@cypress/react18 => @cypress/react` or `@cypress/vue => @cypress/mount-utils`. The bundled code rolls up these dependencies so that they are all in one file and yet the types weren't being rolled. This resulted in `index.d.ts` containing `import {Something} from '@cypress/<something>'` which would be converted to `any` since `@cypress/<something>` doesn't exist once we've moved the packages to the cli package. It also doesn't exist when the package is downloaded from npm since we don't list dependencies that are entirely rolled up inside packageJson.dependencies.

I've replaced the type generation with `rollup-plugin-dts` which provides consistent behavior between rolling up source code and types. If all of the source code for `@cypress/react` gets rolled up when bundling `@cypress/react18`, all of it's types should be rolled up as well.

### Steps to test
In an external repo that has CT setup, checkout the types for `cy.mount(comp, options)`, you should notice the options typed as any for react and vue. Checkout this branch, run `yarn` and then copy `cli/vue` into your external projects `node_modules/cypress` directory, replacing the existing folder and checkout the types again.

### How has the user experience changed?


https://user-images.githubusercontent.com/25158820/198145921-35c875ef-0c48-4b4d-a815-14d966dc505e.mov



### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
